### PR TITLE
Delete the java backend!

### LINKED ISSUE #3167: Delete Java Backend

### PR DESCRIPTION
**Pull request by dwightguth**
This is it! We delete the Java backend from the codebase, remove references to it, clean up the K prelude to remove modules that are no longer needed, delete krun-legacy, and clean up all the places in the code where we were branching on whether or not we were using the Java backend. we also delete a couple command line flags that were only being used by the Java backend.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
...
 ---
 
+K Framework 6.0.0
+=================
+
+Removals
+--------
+
+- Deleted the 'java' backend of K.
+
```
Comment (radumereuta) [inline CHANGELOG.md]: I think we should have a separate PR with just the version bump. There are many other changes that should be included in this changelog.
We should also consider removing `kprove-legacy` before that.

File: kernel/src/main/java/org/kframework/kompile/KompileOptions.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ GlobalOptions getGlobalOptions_UseOnlyInGuiceProvider() {
     public OutputDirectoryOptions outputDirectory = new OutputDirectoryOptions();
 
     // Common options
-    @Parameter(names="--backend", description="Choose a backend. <backend> is one of [llvm|haskell|kore|java]. Each creates the kompiled K definition.")
+    @Parameter(names="--backend", description="Choose a backend. <backend> is one of [llvm|haskell|kore]. Each creates the kompiled K definition.")
```
Comment (radumereuta) [inline kernel/src/main/java/org/kframework/kompile/KompileOptions.java:52]: The `kore` backend doesn't exist.

**Review by radumereuta — APPROVED** [review]
Tested the changes locally and all looks good.

I think we should have a separate PR for the version bump. Other than that, looking good.

File: kernel/src/main/java/org/kframework/kompile/KompileOptions.java (hunk 2 above)
Comment (dwightguth) [reply kernel/src/main/java/org/kframework/kompile/KompileOptions.java:52]: It does, in fact. It emits a kore definition and then immediately stops. It's not really used in practice, but it does exist.

File: CHANGELOG.md (hunk 1 above)
Comment (dwightguth) [reply CHANGELOG.md]: The problem with doing what you suggest is that we are no longer able to say "if you want to use the Java backend, download K 5.5"

File: kernel/src/main/java/org/kframework/kompile/KompileOptions.java (hunk 2 above)
Comment (radumereuta) [reply kernel/src/main/java/org/kframework/kompile/KompileOptions.java:52]: I was looking at `Backends.java` and it doesn't have an option there. I guess this is something else to fix.

File: CHANGELOG.md (hunk 1 above)
Comment (radumereuta) [reply CHANGELOG.md]: Then should we add a minor version bump to cover all the changes until now and then immediately bump it to v6?

Comment (dwightguth) [reply CHANGELOG.md]: That could work, yes. Will you tackle the minor version bump?

Comment (radumereuta) [reply CHANGELOG.md]: I can do that. Either tonight or tomorrow on the train.
Should we remove `kprove-legacy` before v6 as well?

Comment (dwightguth) [reply CHANGELOG.md]: Let's ask about kprove-legacy at the meeting Thursday. I'm not sure how much it's still being used currently.

Comment (radumereuta) [reply CHANGELOG.md]: I made an issue and added it to the board:
https://github.com/runtimeverification/k/issues/3257

**Comment by ehildenb** [comment]
Let's merge!

**Comment by dwightguth** [comment]
It needs conflicts resolved. Feel free to assign someone to do that; I'm hoping to be back to work feeling better from my illness on Monday, but I'm not feeling well enough yet to do it today.

**Comment by radumereuta** [comment]
> It needs conflicts resolved. Feel free to assign someone to do that; I'm hoping to be back to work feeling better from my illness on Monday, but I'm not feeling well enough yet to do it today.

We decided to make a blog post about it and release them at the same time: https://github.com/runtimeverification/k/issues/3290

**Pull request merged**